ARCs/arc-0018.md

```diff
@@ -0,0 +1,348 @@
+            "desc":"Rescinds offered asset",
+            "args":[
+                {"name":"royalty_asset", "type":"asset"}
+            ],
+            "returns":{"type":"void"}
+        },
+        {
+            "name":"move",
+            "desc":"Moves the asset passed from one account to another. This will wipe any offered state from the `from` account",
+            "args":[
+                {"name":"royalty_asset", "type":"asset"}, 
+                {"name":"royalty_asset_amount", "type":"uint64"}, 
+                {"name":"from", "type":"account"},
+                {"name":"to", "type":"account"}
+            ],
+            "returns":{"type":"void"}
+        },
+        {
+            "name":"set_policy",
+            "desc":"Sets the royalty policy for a given app id",
+            "args":[
+                {"name":"royalty_asset", "type":"asset"}, 
+                {"name":"royalty_receiver", "type":"address"},
+                {"name":"royalty_share", "type":"uint64"},
+                {"name":"allowed_asset_1", "type":"asset"},
+                {"name":"allowed_asset_2", "type":"asset"},
+                {"name":"allowed_asset_3", "type":"asset"},
+                {"name":"allowed_asset_4", "type":"asset"}
+            ],
+            "returns":{"type":"void"}
+        },
+        {
+            "name":"transfer",
+            "desc":"Transfers an Asset from one account to another and enforces royalty payments. If payment is in Algos, payment_asset should be 0",
+            "args":[
+                {"name":"royalty_asset", "type":"asset"}, 
+                {"name":"owner_acct", "type":"account"},
+                {"name":"buyer_acct", "type":"account"},
+                {"name":"royalty_receiver", "type":"account"},
+                {"name":"royalty_asset_amount", "type":"uint64"},
+                {"name":"payment", "type":"txn"},
+                {"name":"payment_asset", "type":"asset"}
+            ],
+            "returns":{"type":"void"}
+        }
+    ],
+    "locals": [ 
+        {
+            "name": "{uint64 asset id as big endian bytes}",
+            "desc": "The ID of the asset that has been offered as key, authorizing address as value",
+            "type": "(address,uint64)"
+        }  
+    ],
+    "globals": [ 
+        {
+            "name": "0000000000000000",
+            "desc": "The default royalty policy for any ASA unless its ASA ID is specified",
+            "type": "(address,uint64,uint64,uint64,uint64)"
+        },
+        {
+            "name": "{uint64 asset id as big endian bytes}",
+            "desc": "Royalty policy for asset ID in key ",
+            "type": "(address,uint64,uint64,uint64,uint64)"
+        }  
+    ]
+}
+```
+
+### Metadata
+
+The metadata associated to an asset MAY conform to [ARC-3](./arc-0003.md) or [ARC-69](./arc-0069.md) with an additional field in the `attributes` section specifying the Application ID for the Royalty Enforcer Contract. 
+
+```js
+//...
+"attributes":{
```

Comment:
I think we should use `properties`, not `attributes`.
That would also be better re https://github.com/algorandfoundation/ARCs/pull/61/files

One issue is that ARC-69 only allows simple properties, but maybe we can convince them otherwise.